src/main/java/org/checkerframework/specimin/UnsolvedSymbolVisitor.java

```diff
@@ -544,8 +590,28 @@ public Visitable visit(MethodCallExpr method, Void p) {
   @Override
   public Visitable visit(Parameter parameter, Void p) {
     try {
-      parameter.resolve().describeType();
-      return super.visit(parameter, p);
+      ResolvedParameterDeclaration resolvedParameter = parameter.resolve();
+      // Specimin will update synthetic class based on import statements before working on
+      // parameters. Why a parameter could be resolved but not described? Specimin creates the
+      // synthetic class based on the import statements, so if the synthetic class indeed needs a
+      // placeholder, Specimin will miss it.
+      try {
+        resolvedParameter.describeType();
+        return super.visit(parameter, p);
+      } catch (IllegalArgumentException e) {
+        // following the above explanation, typeName will be in this form path.to.A<path.to.B>
+        String typeName = parameter.getType().asString();
+        @SuppressWarnings(
+            "signature") // since this type is included among the import statements, we can expect
+        // that when it is used, it will be in its simple form.
+        @ClassGetSimpleName String typeToAddPlaceHolder = typeName.substring(0, typeName.indexOf("<"));
```

Comment:
This code should check for commas in the part between the `<`s.

Also, note that this could be recursive: for example, consider the type `List<Map<String, Object>>`. We need to be able to handle arbitrarily-deep type parameters here.

Reply:
As I understand, since this is generic type, we don’t really need to care about how deep it can go.

For example, if this our unsolved type: `ImportDeclaration<List<Map<String, String>> `, then a synthetic declaration such as `ImportDeclaration<T>` should be enough to get the program compuled.